docs/gallery/plot_term_set.py

```diff
@@ -0,0 +1,147 @@
+:py:class:`~hdmf.TermSet` class. The TermSet type
+is experimental and is subject to change in future releases. If you use this type,
+please provide feedback to the HDMF team so that we can improve the structure and
+overall capabilities.
+
+Introduction
+-------------
+The :py:class:`~hdmf.TermSet` class provides a way for users to create their own
+set of terms from brain atlases, species taxonomies, and anatomical, cell, and
+gene function ontologies.
+
+:py:class:`~hdmf.TermSet` serves two purposes: data validation and external reference
+management. Users will be able to validate their data to their own set of terms, ensuring
+clean data to be used inline with the FAIR principles later on.
+The  :py:class:`~hdmf.TermSet` class allows for a reusable and sharable
+pool of metadata to serve as references to any dataset within the NWB ecosystem.
+The :py:class:`~hdmf.TermSet` class is used closely with
+:py:class:`~hdmf.common.resources.ExternalResources` to more efficiently map terms
+to data. Please refer to the tutorial on ExternalResources to see how :py:class:`~hdmf.TermSet`
+is used with :py:class:`~hdmf.common.resources.ExternalResources`.
+
+:py:class:`~hdmf.TermSet` is built upon the resources from LinkmL, a modeling
+language to create YAML schemas, giving :py:class:`~hdmf.TermSet`
+a standardized structure and a variety of tools to help the user manage their references.
+
+How to make a TermSet Schema
+----------------------------
+Before the user can take advantage of all the wonders within the
+:py:class:`~hdmf.TermSet` class, the user needs to a LinkML schema (YAML) that provides
+all the permissible term values. Please refer to https://linkml.io/linkml/intro/tutorial06.html
+to learn more about how LinkML structures their schemas.
+
+1. The name of the schema is up to the user, e.g., the name could be "Species" if the term set will
+   contain species terms.
+2. The prefixes will be the standardized prefix of your source, followed by the URI to the terms.
+   For example, the NCBI Taxonomy is abbreviated as NCBI_TAXON, and Ensemble is simply Ensemble.
+   As mentioned prior, the URI needs to be to the terms; this is to allow the URI to later be coupled
+   with the source id for the term to create a valid link to the term source page.
+3. The schema uses LinkML enumerations to list all the possible terms. Currently, users will need to
+   manually outline the terms within the enumeration's permissible values.
+
+For a clear example, please refer to example_term_set.yaml within the tutorial gallery.
+"""
+######################################################
+# Creating an instance of the TermSet class
+# ----------------------------------------------------
+from hdmf.common import DynamicTable, VectorData
+try:
+    import linkml_runtime  # noqa: F401
+    LINKML_INSTALLED = True
+except ImportError:
+    LINKML_INSTALLED = False
+
+if LINKML_INSTALLED:
+    from hdmf.term_set import TermSet
```

Comment:
If LINKML is strictly optional, then maybe we can add the logic for checking whether TermSets are usable in the `hdmf.term_set` module, e.g., by having a flag `from hdmf.term_set import TERMSET_INSTALLED`.

Reply:
For the tutorials, I'm wondering whether we can just raise an error here if LINKML is not installed and say that this tutorial will not be rendered. Having the `if LINKML_INSTALLED` logic in every cell seems distracting.

Reply:
Wouldn't raising an error always lead to the gallery tests failing on git.